python/plugins/processing/modeler/ModelerParameterDefinitionDialog.py

```diff
@@ -273,13 +279,19 @@ def setupUi(self):
                 self.selector.setCrs(QgsCoordinateReferenceSystem('EPSG:4326'))
             self.verticalLayout.addWidget(self.selector)
 
-        self.verticalLayout.addSpacing(20)
-        self.requiredCheck = QCheckBox()
-        self.requiredCheck.setText(self.tr('Mandatory'))
-        self.requiredCheck.setChecked(True)
-        if self.param is not None:
-            self.requiredCheck.setChecked(not self.param.flags() & QgsProcessingParameterDefinition.FlagOptional)
-        self.verticalLayout.addWidget(self.requiredCheck)
+        elif isinstance(self.param, QgsProcessingDestinationParameter):
+            self.verticalLayout.addWidget(QLabel(self.tr('Default value')))
+            self.defaultWidget = DestinationSelectionPanel(self.param, self.alg, show_encoding=False)
+            self.verticalLayout.addWidget(self.defaultWidget)
+
+        if not isinstance(self.param, QgsProcessingDestinationParameter):
```

Comment:
Note the in 3.0 it's possible to have non mandatory outputs for an algorithm. I think we should expose this to models too - since effectively it's just removing this check and the rest should work...

Reply:
From my point of view, the output is optional if it is optional in the child algorithm that produce the output. So no need to set the optional in the model output.

Reply:
But what if someone wanted to create a model where an normally-optional output is mandatory? E.g. if they wanted to require that users output invalid geometry features to a file. Then it'd be handy to allow the model creator to specify that outputs are mandatory within their own model.